src/main/java/service/operation/SupplyOperation.java

```diff
@@ -0,0 +1,21 @@
+package service.operation;
+
+import data.db.Storage;
+import java.util.Map;
+import model.FruitTransaction;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void operation(FruitTransaction.Operation operation, FruitTransaction fruitTransaction) {
+        if (fruitTransaction.getQuantity() <= 0) {
+            return;
+        }
+        for (Map.Entry<String, Integer> fruit : Storage.fruitsStorage.entrySet()) {
+            if (fruit.getKey().equals(fruitTransaction.getFruit())) {
+                fruit.setValue(fruit.getValue() + fruitTransaction.getQuantity());
+                return;
+            }
+        }
+        Storage.fruitsStorage.put(fruitTransaction.getFruit(), fruitTransaction.getQuantity());
```

Comment:
The current implementation iterates over the entire storage to find a match, which is inefficient. Consider using the Map's get method to check if the fruit exists and then update the quantity. If not, put the new fruit in the storage. This will simplify and optimize the code.